src/main/webapp/resources/js/pages/announcement/components/AnnouncementsPage.jsx

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect, useState } from "react";
+import { PageWrapper } from "../../../components/page/PageWrapper";
+import {
+  getAnnouncements,
+  markAnnouncementRead,
+} from "../../../apis/announcements/announcements";
+import { Avatar, Form, List, notification, Radio, Typography } from "antd";
+import { PriorityFlag } from "./PriorityFlag";
+import { fromNow } from "../../../utilities/date-utilities";
+import ViewReadAnnouncement from "./ViewReadAnnouncement";
+import ViewUnreadAnnouncement from "./ViewUnreadAnnouncement";
+import { blue1 } from "../../../styles/colors";
+
+const { Text } = Typography;
+
+/**
+ * React component to display the page for administration of users.
+ * @returns {*}
+ * @constructor
+ */
+export function AnnouncementsPage({}) {
+  const [announcements, setAnnouncements] = useState([]);
+  const [readToggle, setReadToggle] = useState(false);
+  const [filter, setFilter] = useState("all");
+
+  useEffect(() => {
+    getAnnouncements().then((data) => {
```

Comment:
You should not need to fetch all the announcement again.  On the initial fetch, store them in there own state, and a new `filteredAnnouncements` state, which to use for rendering the list.  When the filter changes just filter the origainal list and set that into the filterAnnouncements.

Reply:
Please see 1a1cf0e53bd7741970ba9a465a118a9b45d3c7ef.